sha1-file.c

```diff
@@ -452,6 +454,52 @@ static int alt_odb_usable(struct raw_object_store *o,
 {
 	struct object_directory *odb;
 
+	if (gvfs_shared_cache_pathname && *gvfs_shared_cache_pathname &&
+	    !strcmp(gvfs_shared_cache_pathname, path->buf)) {
+		/*
+		 * `gvfs.sharedCache` is the preferred alternate that we
+		 * will use with `gvfs-helper.exe` to dynamically fetch
+		 * missing objects.  It is set during git_default_config().
+		 *
+		 * Make sure the directory exists on disk before we let the
+		 * stock code discredit it.
+		 */
+		struct strbuf buf_pack_foo = STRBUF_INIT;
+		enum scld_error scld;
+
+		/*
+		 * Force create the "<odb>" and "<odb>/pack" directories, if
+		 * not present on disk.  Append an extra bogus directory to
+		 * get safe_create_leading_directories() to see "<odb>/pack"
+		 * as a leading directory of something deeper (which it
+		 * won't create).
+		 */
+		strbuf_addf(&buf_pack_foo, "%s/pack/foo",
+			    gvfs_shared_cache_pathname);
+
+		scld = safe_create_leading_directories(buf_pack_foo.buf);
```

Comment:
Good. From reading the commit message, I somehow expected that the directory was created during the config parsing. I think it was this paragraph that put me on the wrong track:

> The above is handled in git_default_config() during the startup sequence.

Reply:
i'll reword it.  thanks!